ci.nix

```diff
@@ -4,14 +4,38 @@
 }:
 let
   self = builtins.getFlake (toString ./.);
-  terraform = builtins.getFlake (toString ./terraform/.);
   inherit (self.inputs.nixpkgs) lib;
   stripDomain = name: lib.head (builtins.match "(.*).nix-community.org" name);
+
+  effects = self.inputs.hercules-ci-effects.lib.withPkgs self.inputs.nixpkgs.legacyPackages.x86_64-linux;
+  terraform-deploy =
+    effects.runIf (src.ref == "refs/heads/trying" || src.ref == "refs/heads/staging")
+      (effects.mkEffect {
+        name = "terraform-deploy";
+        inputs = [ (builtins.getFlake (toString ./terraform/.)).outputs.devShells.x86_64-linux.default.nativeBuildInputs ];
+        src = lib.cleanSource ./.;
+        secretsMap.tf-secrets = "tf-secrets";
+        effectScript = ''
+          export TF_IN_AUTOMATION=1
+          export TF_INPUT=0
+          export SOPS_AGE_KEY="$(readSecretString tf-secrets .SOPS_AGE_KEY)"
+          export TF_TOKEN_app_terraform_io="$(readSecretString tf-secrets .TF_TOKEN_app_terraform_io)"
+
+          pushd terraform
+          terraform init
+          terraform validate
+          if [[ ${src.ref} == "refs/heads/staging" ]]; then
+            terraform apply -auto-approve
+          else
+            terraform plan
```

Comment:
So we would use `bors try` to get a plan?

Reply:
Yes, same as the action.